components/dashboard/src/service/service.tsx

```diff
@@ -191,7 +191,12 @@ export class IDEFrontendService implements IDEFrontendDashboardService.IServer {
         let redirect = false;
         try {
             const desktopLink = new URL(url);
-            redirect = desktopLink.protocol !== "http:" && desktopLink.protocol !== "https:";
+            // allow to redirect only for whitelisted trusted protocols
+            // security: IDE-69
+            redirect =
+                desktopLink.protocol === "vscode:" ||
+                desktopLink.protocol === "vscode-insiders:" ||
+                desktopLink.protocol === "jetbrains-gateway:";
```

Comment:
Looks good, maybe this makes it easier to read?

```suggestion
			const trustedProtocols = ["vscode:", "vscode-insiders:", "jetbrains-gateway:"];
			redirect = trustedProtocols.includes(desktopLink.protocol);
```